tests/testcrate/tests/ui/select_join.stderr

```diff
@@ -1,24 +1,43 @@
 error: mismatched types
 expected type `ForeignKey<_>`
    found type `String`
-  --> $DIR/select_join.rs:38:5
+  --> $DIR/select_join.rs:41:5
    |
-38 |     field1: String,
+41 |     field1: String,
    |     ^^^^^^^^^^^^^^
 ...
-54 |     sql!(Table.all().join(field1, i32_field));
+57 |     sql!(Table.all().join(field1, i32_field));
    |                           ------ in this macro invocation
 
 error: mismatched types
 expected type `ForeignKey<_>`
    found type `i32`
-  --> $DIR/select_join.rs:39:5
+  --> $DIR/select_join.rs:42:5
    |
-39 |     i32_field: i32,
+42 |     i32_field: i32,
    |     ^^^^^^^^^^^^^^
 ...
-54 |     sql!(Table.all().join(field1, i32_field));
+57 |     sql!(Table.all().join(field1, i32_field));
    |                                   --------- in this macro invocation
 
-error: aborting due to 2 previous errors
+error[E0308]: mismatched types
+  --> $DIR/select_join.rs:57:27
+   |
+57 |     sql!(Table.all().join(field1, i32_field));
```

Comment:
I'll have to double-check, but this looks like a bug and should probably not happen.

Reply:
ForeignKey is defined as `pub type ForeignKey<T> = Option<T>;` ,so it is probably  the same error as above, just in less clear form

Reply:
Oh, that's right, it's the same issue as the other.
Perhaps it would make sense to make it a newtype with a Deref to Option (I specified it in #38).